### ISSUE TITLE
Codebase itself is not pydocstyle compliant.

### ISSUE THREAD

**Issue by yanxun827** [PyCQA/pydocstyle#464]
Out of curiosity, I ran `pydocstyle src/` I the pydocstyle codebase and to my surprise there are many errors raised. I would have assumed that a tool being used to check for compliance would strive for compliance itself.

Not sure if this has been raised before or if this is a deliberate decision. If not, I suggest that `pydocstring` should be part of the CI pipeline, so compliance is maintained for every commit.

**Comment by Nurdok** [comment]
PEP257 compliance for pydocstyle is [checked here]( https://github.com/PyCQA/pydocstyle/blob/master/src/tests/test_integration.py#L150). Just running `pydocstyle` on `src/` is not the way to go - we have python files with intentionally invalid code for testing under `src/tests/test_cases/`.

**Comment by yanxun827** [comment]
Thanks for pointing out this is done somewhere as a test. 

I tried to replicate what the test does with `pydocstyle src/pydocstyle --ignore=D104,D105`, but I got at least 90+ errors. 
Examples:
```
src/pydocstyle/violations.py:19 in public class `Error`:
    D203: 1 black line required before class doctoring (found 0)
...
```

Did I miss out on something? Or are these somehow deliberate.

**Comment by Nurdok** [comment]
@yanxun827 - you're right! I looked again into the test code and there was a bug in the test itself. I fixed both the test and violations in #465.

Thanks for not giving up and keeping us honest!
